File: frontend/src/api/comment.ts
import request from '@/utils/request'

export interface Comment {
  id: number
  videoId: number
  userId: number
  content: string
  createTime: string
  nickname?: string
  avatar?: string
}

export function addComment(data: { videoId: number, content: string }) {
  return request({
    url: '/comments',
    method: 'post',
    data
  })
}

export function listComments(videoId: number) {
  return request({
    url: '/comments',
    method: 'get',
    params: { videoId }
  })
}

export function deleteComment(id: number) {
  return request({
    url: `/comments/${id}`,
    method: 'delete'
  })
}

File: backend/src/main/java/com/combine/auditorium/entity/Comment.java
package com.combine.auditorium.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("video_comment")
public class Comment {
    @TableId(type = IdType.AUTO)
    private Long id;
    
    private Long videoId;
    private Long userId;
    private String content;
    private LocalDateTime createTime;
    
    // 非数据库字段，用于关联查询用户信息
    @TableField(exist = false)
    private String nickname;
    @TableField(exist = false)
    private String avatar;
}

